### PR TITLE
Added support for CLI variables

### PR DESCRIPTION
**Pull request by kunagpal**
This pull request adds support for the inclusion of global variables via the CLI, addresses #1013.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -154,6 +154,10 @@ newman.run({
   Specify whether or not to force the unicode disable option. When supplied, all symbols in the output will be replaced
   by their plain text equivalents.
 
+- `--global-var <key>=<value>`<br />
+  Allows the specification of global variables via the command line, in a key=value format. Multiple CLI global variables
+  can be added by using `--global-var` multiple times, like so: `--global-var foo=bar --global-var alpha=beta`.
```
Comment (shamasis) [inline README.md]: Add quotations in docs by default?

File: lib/cli/index.js [hunk 2]
```diff
@@ -312,9 +312,26 @@ var _ = require('lodash'),
...
+             * also supported.
+             *
+             * @param {String} arg - The argument provided to `--global-var`.
+             * @returns {{key, value}} - The object representation of the current CLI variable.
+             * @throws {Error} - An error is thrown for arguments that are not in the form foo=bar.
+             */
+            cliVar = function (arg) {
+                var eqIndex = arg.indexOf('=');
+
+                if (eqIndex === -1) { throw new Error('CLI global vars must be specified in a `key=value` format'); }
```
Comment (shamasis) [inline lib/cli/index.js]: We can make `eq` optional and define the var with undefined value. Would negate the need to throw error. Later we anyway plan to augment CLI reporter with summary of environemnt and globals uses, that would aide debugging.

File: lib/cli/index.js [hunk 3]
```diff
@@ -312,9 +312,26 @@ var _ = require('lodash'),
...
+
+                // This is done instead of splitting by `=` to avoid chopping off `=` that could be present in the value
+                return { key: arg.slice(0, eqIndex), value: arg.slice(eqIndex + 1) };
```
Comment (shamasis) [inline lib/cli/index.js]: 👍

File: lib/cli/index.js [hunk 4]
```diff
@@ -398,6 +415,12 @@ var _ = require('lodash'),
             defaultValue: false
         });
 
+        runParser.addArgument(['--global-var'], {
+            help: 'Allows the specification of global variables via the command line, in a key=value format',
+            type: cliVar,
+            action: 'append'
```
Comment (shamasis) [inline lib/cli/index.js]: 👍

File: README.md (hunk 1 above)
Comment (kunagpal) [inline README.md]: Done

File: lib/cli/index.js (hunk 2 above)
Comment (kunagpal) [inline lib/cli/index.js]: Done

**Comment by czardoz** [comment]
Would be good to have an integration test for this change (a CLI integration test)

**Comment by kunagpal** [comment]
@czardoz Yes, I've added CLI tests :)

**Pull request merged**